### PR TITLE
Add human-friendly default output with auto-detection

### PR DESCRIPTION
**Pull request by avelino**
Every command dumped raw JSON, which is great for scripts but terrible for humans typing in a terminal.

Output now auto-detects context: interactive terminal gets colored tables (comfy-table + console) and a spinner during async operations (indicatif), piped/redirected output stays JSON for backwards compatibility. A --json flag forces JSON in any context. Tests go from 55 to 77 covering all format paths, content types and edge cases. CLI and scripting docs updated.

fixed: #21

### CODE REVIEW

File: docs/reference/cli.md [hunk 1]
```diff
@@ -10,29 +24,65 @@ Show usage information.
...
+3 server(s) configured
+```
+
+JSON output (`--json` or piped):
+
 ```json
 [
-  { "name": "sentry", "type": "http", "url": "https://mcp.sentry.dev/sse" },
+  { "name": "sentry", "type": "http", "url": "https://mcp.sentry.dev/mcp" },
   { "name": "slack", "type": "stdio", "command": "npx", "args": ["-y", "slack-mcp-server@latest"] }
```
[the remark this answers is not in the document]
Comment (avelino) [reply docs/reference/cli.md]: Good catch. Reverted both occurrences back to `https://mcp.sentry.dev/sse` to stay consistent with README.md, tutorial, configuration guide, and all other docs.

File: src/spinner.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        );
+        bar.set_message(message.to_string());
+        bar.enable_steady_tick(Duration::from_millis(80));
+        Self { bar }
+    }
+
+    pub fn stop(self) {
+        self.bar.finish_and_clear();
+    }
+}
```
[the remark this answers is not in the document]
Comment (avelino) [reply src/spinner.rs]: Fixed both issues:

1. Added `impl Drop for Spinner` that calls `finish_and_clear()`, so early `?` returns clean up the spinner automatically.
2. `Spinner::start()` now checks `std::io::stderr().is_terminal()` — when not a TTY, it uses `ProgressBar::hidden()` to suppress all output.

File: src/output.rs [hunk 3]
```diff
@@ -99,27 +157,670 @@ pub fn print_search_results(servers: &[RegistryServer]) -> Result<()> {
...
+                    let mime = content.mime_type.as_deref().unwrap_or("unknown");
+                    println!("{}", style(format!("[resource: {mime}]")).dim());
+                }
+            }
+            other => {
+                if let Some(ref text) = content.text {
+                    println!("{}", text);
+                } else {
+                    println!("{}", style(format!("[{other}: unsupported content type]")).dim());
+                }
```
[the remark this answers is not in the document]
Comment (avelino) [reply src/output.rs]: Fixed together with the Sentry comment above. Introduced a local `out\!` macro that routes to `eprintln\!`/`println\!` based on `is_error`, applied to all content branches uniformly. Empty error content now emits `"(no details)"` as fallback.
